include/core.hpp

```diff
@@ -33,7 +33,10 @@ namespace color_coded
 
     inline std::string temp_dir()
     {
-      static auto const dir(fs::temp_directory_path() / "color_coded/");
+      static auto const temp_dir(fs::temp_directory_path());
+      static auto const user_dir(std::string("color_coded-") +
+                                 std::to_string(geteuid()));
```

Comment:
```c++
static auto const user_dir("color_coded-" + std::to_string(geteuid()) + "/");
```

The extra `std::string` rvalues are needless, since `std::string` has an `operator +` overload where a C string is on the left-hand side ("color_coded-" + UID string) which returns a `std::string` to then call `operator +` with a C string on the right-hand side. This minimizes the number of string temporaries made and is also less typing.

I opted to choose the explicit concatenation of "/" precisely because it's explicit. No matter the language, there's added safety and maintainability in explicit saying what you want.